### ISSUE TITLE
Blurry Image Outputs with Particles

### ISSUE THREAD

**Issue by fzehracetin** [Pranshu258/Image_Completion#3]
Hi,

First of all, thank you for work and all codes that you share. I'm trying to work with your method. The outputs for Image Completion always blurry and they have particles. 

This image is very similar to the example in the paper. I worked with a square mask in the center of the image. Here is the output:

![resized_Complete](https://user-images.githubusercontent.com/37731704/81791979-186c7d80-9510-11ea-880c-19edb3e8ef83.png)

I  put some of my outputs to this Drive folder if you want to see: https://drive.google.com/drive/folders/1t9h8f8K4qkP4PT6hF9gOzwizDBXFQzkp?usp=sharing

So, I'm curious. Am I missing something or the outputs are normal for this method?

**Comment by Pranshu258** [comment]
Thanks for referring to our repository for your work. We tried to remain as close to paper as possible. But with the resources that were known to us at the time, we were not able to replicate the results exactly as shown in the paper. 
We also got results similar to yours. For very simple images, the results are decent, but in images with many patterns like the one above, there is noise. 
Yes, there is scope of improvement, but we were not able to figure out the issue in our implementation, given the time constraints we were under. I suspect it has something to do with the way we have implemented the energy optimization step.
